types/react-sizes/index.d.ts

```diff
@@ -0,0 +1,18 @@
+// Type definitions for react-sizes 2.0
+// Project: https://github.com/renatorib/react-sizes#readme
+// Definitions by: janKir <https://github.com/janKir>, Micah Stubbs <https://github.com/micahstubbs>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.5
+
+export interface Sizes {
+    width: number;
+    height: number;
+}
+
+export type mapSizesToProps<SP extends object> = (sizes: Sizes) => SP;
```

Comment:
Nitpick: I would inline this type in `withSizes` below. It’s unusual to have a type parameter (`SP`) that doesn’t appear in a function parameter position (it’s only in the return type here). In actual fact, it makes sense when combined with `withSizes` because `SP` becomes the `SP` from there, and _does_ wind up in the parameters of `withSizes`. But if you took the type `mapSizesToProps` on its own, it doesn’t make as much sense.

(Other nitpick is that type names are normally PascalCased, but that’s a moot point if you inline it.)

Reply:
thanks for the review, will take a closer look. 🔍

Reply:
added PascalCase

Reply:
will try to inline now

Reply:
ok, had a go at inlining the `mapSizesToProps` function's type definitions

Reply:
now we are looking at: 

```typescript
export interface Sizes {
    width: number;
    height: number;
}

export function WithSizes<SP extends object, P extends SP>(
    mapSizesToProps: (sizes: Sizes) => SP;
): (component: React.ComponentType<P>) => React.ComponentType<P>;
```